agent/consul/leader_connect.go

```diff
@@ -861,6 +869,8 @@ func retryLoopBackoff(ctx context.Context, loopFn func() error, errFn func(error
 			case <-timer.C:
 				continue
 			}
+		} else if err == nil && abortOnSuccess {
```

Comment:
I guess technically the first part is unnecessary
```suggestion
		} else if abortOnSuccess {
```